### PR TITLE
[ISBN Verifier] Error in tests

### PR DESCRIPTION
**Pull request by Keftcha**
Check if the `X` character is only the verification digit

PR according to [this](https://forum.exercism.org/t/isbn-verifier-error-in-tests-that-check-if-the-x-character-is-only-the-verification-digit/19639) topic.

Resolve exercism/python#4014

### CODE REVIEW

File: exercises/isbn-verifier/canonical-data.json [hunk 1]
```diff
@@ -61,13 +61,31 @@
     },
     {
       "uuid": "28025280-2c39-4092-9719-f3234b89c627",
-      "description": "X is only valid as a check digit",
```
Comment (Cool-Katt) [inline exercises/isbn-verifier/canonical-data.json:64]: Tests are immutable, so you shouldn't touch the already existing test cases. 

Take a look at the [guide](https://github.com/exercism/problem-specifications?tab=readme-ov-file#changing-tests) for more info.

**Review by Cool-Katt — CHANGES_REQUESTED** [review]

File: exercises/isbn-verifier/canonical-data.json [hunk 2]
```diff
@@ -68,6 +68,37 @@
       },
       "expected": false
     },
+    {
+      "uuid": "0153f517-3008-4832-a168-8155c2ec0d94",
+      "reimplements": "28025280-2c39-4092-9719-f3234b89c627",
+      "description": "X is only valid as a check digit - X character is substituted for 10 and multiplied",
```
Comment (Cool-Katt) [inline exercises/isbn-verifier/canonical-data.json]: Do we need to keep this test at all of its superseded by the other one that also catches the edge cases? 

If yes, then I'd vote to change the names to something more distinct for the new test, so we can avoid the reimplementation. (I don't have a good name suggestion at the moment, sry)

File: exercises/isbn-verifier/canonical-data.json [hunk 3]
```diff
@@ -68,6 +68,37 @@
...
+      ],
+      "property": "isValid",
+      "input": {
+        "isbn": "3-598-2X507-9"
+      },
+      "expected": false
+    },
+    {
+      "uuid": "fdb14c99-4cf8-43c5-b06d-eb1638eff343",
+      "description": "X is only valid as a check digit - X character is substituted for 10 and not multiplied",
```
Comment (IsaacG) [inline exercises/isbn-verifier/canonical-data.json]: I'm not convinced this test is needed.

File: exercises/isbn-verifier/canonical-data.json [hunk 4]
```diff
@@ -68,6 +68,37 @@
...
+      "property": "isValid",
+      "input": {
+        "isbn": "3-598-2X507-5"
+      },
+      "expected": false
+    },
+    {
+      "uuid": "47fa54cb-507e-45d3-901a-096be447dfd9",
+      "description": "X is only valid as a check digit - X character is ignored altogether",
```
Comment (IsaacG) [inline exercises/isbn-verifier/canonical-data.json]: I'm not convinced this test is needed.

File: exercises/isbn-verifier/canonical-data.json (hunk 2 above)
Comment (IsaacG) [reply exercises/isbn-verifier/canonical-data.json]: If we're not adding the other tests, this description doesn't need to change.

File: exercises/isbn-verifier/canonical-data.json [hunk 5]
```diff
@@ -68,6 +68,37 @@
       },
       "expected": false
     },
+    {
+      "uuid": "0153f517-3008-4832-a168-8155c2ec0d94",
+      "reimplements": "28025280-2c39-4092-9719-f3234b89c627",
+      "description": "X is only valid as a check digit - X character is substituted for 10 and multiplied",
+      "comments": [
+        "Rename this test due to multiple tests on X that should be only valid as a check digit"
```
Comment (IsaacG) [inline exercises/isbn-verifier/canonical-data.json]: This comment looks like a commit comment. Looking at the test data and not the changes, this comment isn't useful so it should not be here.

File: exercises/isbn-verifier/canonical-data.json (hunk 2 above)
Comment (oxe-i) [reply exercises/isbn-verifier/canonical-data.json]: > Do we need to keep this test at all of its superseded by the other one that also catches the edge cases?

I've commented in the forum topic, but one does not supersede the other. The current one checks when X is substituted for 10 and multiplied by the position (as the other digits are). The proposed one checks when X is substituted for 10 without being multiplied by the position, but fails to assert when it is multiplied by the position. So, they are complementary tests.

File: exercises/isbn-verifier/canonical-data.json (hunk 3 above)
Comment (oxe-i) [reply exercises/isbn-verifier/canonical-data.json]: > I'm not convinced this test is needed.

This is the test proposed in the topic. The other one is the current test already present in the specs, only renamed.

File: exercises/isbn-verifier/canonical-data.json (hunk 5 above)
Comment (Keftcha) [reply exercises/isbn-verifier/canonical-data.json]: Is something like *«Add a more precise description»* better ?

Comment (IsaacG) [reply exercises/isbn-verifier/canonical-data.json]: For a changelog? Sure. For the canonical data? There's no need for the canonical data file to contain change history. That's what the git log is for. The comments in the file should be useful to explain the data in the file. The comments in the file should _not_ be used to explain the history of the file.

Comment (Keftcha) [reply exercises/isbn-verifier/canonical-data.json]: I don't really know what to put in comments as I just rename the test. The [guide](https://github.com/exercism/problem-specifications?tab=readme-ov-file#changing-tests) specifically tell that I should add a comment.

> Re-implemented test cases must use the comments field to explain why a test case was re-implemented

This is what I try to do by explaining why I've renamed the test.  
Do you have an idea of what a correct comment could be for this case ?

Comment (Cool-Katt) [reply exercises/isbn-verifier/canonical-data.json]: In that case, I suggest not touching the original test and instead just come up with a better name to the new one. Seems like the simplest option to me, even if it's not the most clear one.

**Review by IsaacG — APPROVED** [review]
+cc @ErikSchierboom  for write approval

**Comment by Keftcha** [comment]
Thanks you !
